codec_test.go

```diff
@@ -28,51 +29,51 @@ func init() {
 	}
 }
 
-func TestCompressDecompress(t *testing.T) {
+func TestCompressDecompressEmpty(t *testing.T) {
+	testCompressDecompress(t, []byte{})
+}
+
+func TestCompressDecompressSmallText(t *testing.T) {
 	testCompressDecompress(t, []byte("hello"))
 }
 
 func TestCompressDecompressManySizes(t *testing.T) {
-
-	// Pick two boundary value sizes to test with
-	sizesToTest := []int{1, 65535}
-
-	// And 1000 random sizes to test with
-	for i := 0; i < 1000; i++ {
-		randomSize := rando.Intn(65535)
-		if randomSize > 1 {
-			sizesToTest = append(sizesToTest, randomSize)
-		}
-	}
-
-	// Test compress/decompress loop with all of those sizes
-	for _, sizeToTest := range sizesToTest {
-		testCompressDecompress(t, compressibleDataOfLength(sizeToTest))
+	for baseSize := 0; baseSize < 65536; baseSize += 256 {
+		t.Run(fmt.Sprintf("%d-%d", baseSize, baseSize+255), func(t *testing.T) {
+			for sizeLoop := baseSize; sizeLoop < baseSize+256; sizeLoop++ {
+				size := sizeLoop // make a copy that doesn't change, to use inside the function
+				t.Run(fmt.Sprintf("%d", size), func(t *testing.T) {
```

Comment:
This approach, in conjunction with the `-v` flag, creates a bit too much output.

See http://drone.couchbase.io/couchbase/go-blip/48

I think we should either remove the -v flag from `drone.yml`, or rework this to use less subtests (actually it's using doubly nested subtests..).   As it stands, it looks like there are still 65536 subtests being run.  

What if there were only 256 subtests to get the parallelism and the nested subtests were removed?   This should make the logs much more manageable. 

The `-v` flag is just useful for seeing which tests ran as part of the build, and log output for successful tests.  It's really useful for seeing the log output in context -- for example blip errors are alarming in the logs, unless it's within a context of `TestBlipError` (for example)

As far as total execution time, it's 391.924s on the drone CI machine, which is a bit long for such a small library, but still within range.  I'm not sure if the parallelism is actually adding much there, since the VM only has one or two cpus.  But if we later upgrade the box, it would have a positive effect.

Reply:
You're right; I'll change it to 256 subtests (and do the logContext change.)

Reply:
In my local testing, that's working *a lot* better.  (no longer blows up my IDE) Thanks!